app/models/condition.rb

```diff
@@ -8,7 +8,10 @@ class Condition < ApplicationRecord
   has_one :form, through: :routing_page
 
   def save_and_update_form
-    save && form.update!(question_section_completed: false)
+    save
+    form.update!(question_section_completed: false)
+    form.draft_new_live_form! if form.live?
+    form.create_draft_from_archived_form! if form.archived?
```

Comment:
not going to need these changes if user triggers the state change themselves and specifically "creates a new draft"